app.js

```diff
@@ -0,0 +1,33 @@
+const express = require('express');
+const cors = require('cors');
+const port = 3005;
+const app = express();
+app.use(cors());
+
+let animalsArr = ["lion", "zebra", "tiger", "elephant", "shark", "giraffe", ""];
+
+const isAnimal = (req, res, next) => {
+    console.log(req.params);
+    if(animalsArr.includes(req.params.name)){
+        res.json({
+            status: "Success",
+            message: true,
+        })
+    } else {
+        res.json({
+            status: "Failed",
+            message: false,
+        })
+    }
+}
+
+app.get("/animal", (req, res, next) => {
+    res.json({animalsArr});
+})
+
+app.get("/animal/:name", isAnimal, (req, res) => {
+    res.json(animalsArr);
```

Comment:
You wouldnt need to send a res from here since your middleware function would already have sent a return if the animal was included and not included.